openmc/geometry.py

```diff
@@ -65,14 +63,22 @@ def get_offset(self, path, filter_offset):
 
         """
 
+        # Find the distribcell index of the cell.
+        cells = self.get_all_cells()
+        if path[-1] in cells:
+            distribcell_ind = cells[path[-1]].distribcell_ind
```

Comment:
I agree that `distribcell_ind` isn't a great name but I can't figure out a better one.  `instance_id` isn't quite right for this variable.  There's a bit of confusion here because `offset` is an overloaded term that can mean one of two things.  In the context of `lattice % offset` and `universe % offset`, it means something like "next instance of a particular distributed cell".  In the context of `filter % offset` (or `cell % distribcell_ind` in this PR) it's kind of like "which distributed cell are we talking about".